lib/trino-record-decoder/src/test/java/io/trino/decoder/avro/TestAvroDecoder.java

```diff
@@ -369,6 +373,14 @@ public void testSchemaEvolutionToIncompatibleType()
                 .hasMessageMatching("Decoding Avro record failed.");
     }
 
+    @Test
+    public void testLongDecodedAsTimestampMillis()
+    {
+        DecoderTestColumnHandle row = new DecoderTestColumnHandle(0, "row", TIMESTAMP_MILLIS, "id", null, null, false, false, false);
+        Map<DecoderColumnHandle, FieldValueProvider> decodedRow = buildAndDecodeColumn(row, "id", "{\"type\":\"long\",\"logicalType\":\"timestamp-millis\"}", 1658463479L);
```

Comment:
``` 
GenericData.get().addLogicalTypeConversion(new TimeConversions.TimestampMillisConversion());
```
TypeConversion can be added using similar snippets.

Reply:
> Avro serializes/de-serializes it as long but as per specification it looks like for Java we would pass the input as Instant for Timestamp logical type

Yes. To support many and many cases, there's a lot of work to do. I don't think one PR should handle all of it or you can try it out. I'd say the patch here can resolve the specific issue of bumping trino version for Pulsar Trino Plugin.

> GenericData.get().addLogicalTypeConversion(new TimeConversions.TimestampMillisConversion());

What does it mean? If I get it right you suggest one more possible case avro timestamp can be represented as. Then my answer is as above.

Reply:
Yeah I agree but I'm afraid if this is the correct approach. One major thing which bugs me is that we manually do the conversion from a long to a micro-second precision compatible format - but ideally it should happen as a part of Avro library and it does support it. And one additional benefit is that in case if Avro modifies their internal representation it shouldn't affect us.

Reply:
> but ideally it should happen as a part of Avro library and it does support it

Can you show me how Avro "does support it"? I'm glad to adapt such a way.

Reply:
This method shows on how Avro implements it 
https://github.com/apache/avro/blob/a2c111dc3581f9549b173f7c6a9073cfbf173558/lang/java/avro/src/test/java/org/apache/avro/generic/TestGenericLogicalTypes.java#L368

Reply:
Thank you! I'm looking into it this week :)

Reply:
@Praveen2112 after a review I think the conversion is inevitable because Trino has it's own representation, introduced by:

* https://github.com/trinodb/trino/commit/21e3ddf55a641e1c17b7e4a17689223ff1120b42
* https://github.com/trinodb/trino/commit/19811d3d58e908d145639b1975ec111a0002e138

I take a look at the TimeConvertions, but it seems to be used in code, not during deserialization. Pulsar Trino Plugin users meet a use case that the actual payload _is_ int/long according to the logical type, and there's no `Instant` or `LocalTime` to be converted.

I don't have a good idea to make further revision on this codepath. You're welcome to show a patch.